src/components/SVGViewer/SVGViewer.tsx

```diff
@@ -104,8 +121,21 @@ export class SVGViewer extends React.Component<SvgViewerProps, SvgViewerState> {
       this.setCustomClasses();
     }
 
-    if (prevProps.documentId !== this.props.documentId) {
-      this.presetSVG();
+    if (isDocumentIdProps(prevProps)) {
+      if (
+        isDocumentIdProps(this.props) &&
+        this.props.documentId !== prevProps.documentId
+      ) {
+        this.presetSVG();
+      } else if (isDocumentIdProps(this.props)) {
+        this.presetSVG();
```

Comment:
but you do `this.presetSVG();` both on if and `else`.
why do you need to have them separately?

Reply:
I made the logic cleaner with comments